### PR TITLE
Refactor(eos_cli_config_gen): schema review and template organization

### PR DESCRIPTION
**Pull request by carlbuchmann**
## Change Summary

- Remove convert_dicts from all eos_cli_config_gen templates
- Where possible, rename templates to root key matching name of schema fragment
  exceptions:
   mac-address-table-aging-time (mac_address_table.aging_time)- split out due to placement in cli
   mac-address-table-notification (mac_address_table.aging_time.notification) - split out due to placement in cli
   policy-maps-pbr (policy_maps.pbr) - split out for documentation purpose
   policy-maps-qos  (policy_maps.qos)  - split out for documentation purpose

## Component(s) name

`arista.avd.eos_cli_config_gen`

## How to test

CI Passes, and no changes to molecule scenarios

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2 [hunk 1]
```diff
@@ -53,23 +53,23 @@
...
-{% include 'eos/domain-lookup.j2' %}
+{% include 'eos/ip-domain-lookup.j2' %}
 {# name servers #}
-{% include 'eos/name-servers.j2' %}
+{% include 'eos/name-server.j2' %}
 {# dns domain #}
 {% include 'eos/dns-domain.j2' %}
 {# domain-list #}
 {% include 'eos/domain-list.j2' %}
 {# Object Tracking #}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2:71]: any wish to also sanitize the comments or we can ignore for now?

**Review by gmuloc — APPROVED** [review]
LGTM given that the pipeline succeeds without change and the renaming makes sense. only question is whether or not you would want to make the comments matching as well as some of them may be a bit out of sync but this is cosmetic.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2 (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2:71]: Good catch, I will clean up the comments!

**Review by ClausHolbechArista — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2 (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2:71]: updated in 2f4128a

**Pull request merged**